libs/designsystem/src/lib/components/calendar/calendar.component.spec.ts

```diff
@@ -229,6 +241,127 @@ describe('CalendarComponent', () => {
     ).toEqual('M T W T F S S');
   });
 
+  describe('active month', () => {
+    let todayDate: Date;
+
+    beforeEach(() => {
+      todayDate = new Date(2021, 0, 1);
+      spectator.setInput('todayDate', todayDate);
+    });
+
+    describe('when `minDate` is set', () => {
+      let minDate: Date;
+
+      beforeEach(() => {
+        minDate = new Date(2017, 0, 1);
+        spectator.setInput('minDate', minDate);
+      });
+
+      describe('when `minDate` changes to a later date', () => {
+        let newMinDate: Date;
+
+        beforeEach(() => {
+          newMinDate = new Date(2021, 11, 30);
+          spectator.setInput('minDate', newMinDate);
+        });
+
+        it('should set active month based on changed `minDate`', () => {
+          expect(spectator.component.activeMonthName).toEqual(moment(newMinDate).format('MMMM'));
+        });
+      });
+    });
+
+    describe('when `maxDate` is set', () => {
+      let maxDate: Date;
+
+      beforeEach(() => {
+        maxDate = new Date(2023, 0, 1);
+        spectator.setInput('maxDate', maxDate);
+      });
+
+      describe('when `maxDate` changes to an earlier date', () => {
+        let newMaxDate: Date;
+
+        beforeEach(() => {
+          newMaxDate = new Date(2020, 6, 30);
+          spectator.setInput('maxDate', newMaxDate);
+        });
+
+        it('should set active month based on changed `maxDate`', () => {
+          expect(spectator.component.activeMonthName).toEqual(moment(newMaxDate).format('MMMM'));
+        });
+      });
+    });
+  });
+
+  describe('year navigator', () => {
+    describe('by default', () => {
+      it('should not render', () => {
+        expect(spectator.element.querySelector('kirby-dropdown')).toBeNull();
+      });
+    });
+
+    describe('when yearNavigatorOptions are set', () => {
+      let todayDate: Date;
+      let yearNavigatorOptions: CalendarYearNavigatorConfig;
+
+      beforeEach(() => {
+        todayDate = new Date(2021, 0, 1);
+        yearNavigatorOptions = { from: -3, to: 2 };
+
+        spectator.setInput('todayDate', todayDate);
+        spectator.setInput('yearNavigatorOptions', yearNavigatorOptions);
+      });
+
+      it('should render', () => {
+        expect(spectator.element.querySelector('kirby-dropdown')).not.toBeNull();
+      });
+
+      // TODO: Should this test be split up into 2 tests or is it acceptable to keep the expects in the same test?
+      it('should change year on navigation', () => {
```

Comment:
Should this test be split up into 2 tests or is it acceptable to keep the expects in the same test? it is both testing navigatedYear and the actual change in the dropdown button text.